script.js

```diff
@@ -12,6 +12,116 @@ const headerListElement = document.getElementById('header_list');
 const buttonElement = document.getElementById('button_back');
 let menuIsActive = false;
 let submenuIsActive = false;
+const buttonRight = document.getElementById('button_right');
+const buttonLeft = document.getElementById('button_left');
+const carouselText = document.getElementById('carousel_text');
+const carouselBox = document.getElementById('carousel_box');
+const carouselElement = document.getElementById('carousel');
+
+let carouselTextObject = document.createDocumentFragment();
+carouselTextObject = carouselText;
+
+let images = Array.from(carouselBox.children);
+let slider = [];
+images.forEach(function(item, i, images) {
+    slider[i] = images[i].src; 
+    item.remove();
+}); 
+
+let step = 0;
+let carousel = Array.from(carouselBox.children);
+function carouselObject (position, location) {
+    let objectDiv = document.createElement('div');
+    let objectImg = document.createElement('img');
+    objectImg.src = slider[position];
+    objectImg.classList.add('carousel__object');
+    objectDiv.classList.add('carousel__box');
+    objectDiv.classList.add(location);
+
+    if (position === 0) {
+        objectDiv.appendChild(carouselTextObject);
+    }
+    objectDiv.appendChild(objectImg);
+    carouselElement.appendChild(objectDiv);
+    carousel.push(objectDiv);
+}
+function drawCarouselSlides() {
+    carouselObject(0, 'middle');
+    carouselObject(1, 'right');
+    carouselObject (slider.length - 1, 'left');
+}
+drawCarouselSlides();
+
+function drawCarouselSlideRight() {
+    let position = step - 1;
+    if (position < 0) {
+        position = slider.length - 1;
+    }
+    carouselObject (position, 'right');
+    step++;
+    if (step === slider.length) {
+        step = 0;
+    }
+};
+
+function drawCarouselSlideLeft() {
+    let position = step + 1;
+    if (position === slider.length) {
+        position = 0;
+    }
+    carouselObject(position, 'left');
+    step--;
+    if (step < 0) {
+        step = slider.length - 1;
+    }
+};
+
+function throttle (callback, limit) {
+    let waiting = false;                     
+    return function () {                                     
+         if (!waiting) {                   
+            callback.apply(this, arguments);  
+            waiting = true;                   
+            setTimeout(function () {          
+                waiting = false;             
+            }, limit);
+        }
+    }
+}
+
+buttonRight.addEventListener('click', throttle(function (){
+    carousel.forEach(function(item, images) { 
+        if (item.className === 'carousel__box left'){
+            item.remove();
+        }
+    })
+    carousel.forEach(function(item, images) {
+        if (item.className === 'carousel__box middle') {
+            item.className = 'carousel__box left';
+        }
+        else {
+            item.className = 'carousel__box middle';
+        }
+    })
+    drawCarouselSlideRight();
+}, 500));
+
+buttonLeft.addEventListener('click', throttle(function (){
```

Comment:
Still contains anonymous function, there is no reason to use it here.

Reply:
Separated functions and EventListeners